http/obmc_shell.hpp

```diff
@@ -0,0 +1,245 @@
+#pragma once
+
+#include <pty.h>
+#include <pwd.h>
+#include <termios.h>
+
+#include <app.hpp>
+#include <boost/asio/posix/stream_descriptor.hpp>
+#include <boost/process/io.hpp>
+#include <websocket.hpp>
+
+#include <map>
+
+namespace crow
+{
+
+namespace obmc_shell
+{
+
+class Handler : public std::enable_shared_from_this<Handler>
+{
+  public:
+    Handler(crow::websocket::Connection* conn) :
+        session(conn), streamFileDescriptor(conn->getIoContext()),
+        doingWrite(false)
+    {}
+
+    ~Handler() = default;
+
+    void doClose()
+    {
+        streamFileDescriptor.close();
+
+        // boost::process::child::terminate uses SIGKILL, need to send SIGTERM
+        int rc = kill(pid, SIGKILL);
+        session = nullptr;
+        if (rc)
+        {
+            return;
```

Comment:
If kill returns a non-zero value, should we just return? If fail to kill process, then will we leave a dangling process?

Reasons why kill will return non-zero value.

ERRORS         top
       EINVAL An invalid signal was specified.

       EPERM  The calling process does not have permission to send the
              signal to any of the target processes.

       ESRCH  The target process or process group does not exist.  Note
              that an existing process might be a zombie, a process that
              has terminated execution, but has not yet been wait(2)ed
              for.